libdebug/debugger/internal_debugger.py

```diff
@@ -1257,7 +1257,7 @@ def __threaded_wait(self: InternalDebugger) -> None:
             self.resume_context.resume = True
             self.debugging_interface.wait()
             if self.resume_context.resume:
-                self.debugging_interface.cont()
+                self.debugging_interface.cont(self.resume_context.thread_to_cont)
```

Comment:
Are you sure that `self.resume_context.thread_to_cont` is not always `None` at this point? The wait is executed from the queue after the continue, and continue, as its last operation, sets `self.resume_context.thread_to_cont = None`.